gpkit/substitution.py

```diff
@@ -179,8 +178,7 @@ def substitution(nomial, substitutions, val=None):
                 sub = VarKey(name=sub, **descr)
                 exps_[i] += HashVector({sub: x})
                 varlocs_[sub].append(i)
-            elif isinstance(sub, VarKey):
-                sub = VarKey(sub)
+            elif isinstance(sub, (VarKey, Monomial)):
                 if isinstance(var.units, Quantity):
                     try:
                         new_units = var.units/sub.units
```

Comment:
When was the last time you looked at the coverage information of your unit
tests?

On Mon, 07 Mar 2016, 23:43 Warren Hoburg <notifications@github.com> wrote:

> In gpkit/substitution.py
> <https://github.com/hoburg/gpkit/pull/504#discussion_r55315772>:
>
> > @@ -179,8 +178,7 @@ def substitution(nomial, substitutions, val=None):
> >                  sub = VarKey(name=sub, **descr)
> >                  exps_[i] += HashVector({sub: x})
> >                  varlocs_[sub].append(i)
> > -            elif isinstance(sub, VarKey):
> > -                sub = VarKey(sub)
> > +            elif isinstance(sub, (VarKey, Monomial)):
> >                  if isinstance(var.units, Quantity):
> >                      try:
> >                          new_units = var.units/sub.units
>
> It's shocking that we've had this wrong for so long -- we should add some
> more unit tests to make sure it remains right. This is when I agree with
> pylint on too many branches being a bad thing -- we need to test every
> branch.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hoburg/gpkit/pull/504/files#r55315772>.
>